### PR TITLE
fix external plugin offsets, better types, new tests

### PR DESCRIPTION
**Pull request by nhanphan**

### CODE REVIEW

**Review by blockiosaurus — APPROVED** [review]

File: clients/js/test/externalPlugins/oracle.test.ts [hunk 1]
```diff
@@ -1161,6 +1412,24 @@ test('it can use preconfigured owner pda oracle to deny burn', async (t) => {
...
+        authority: {
+          type: 'UpdateAuthority',
+        },
+        baseAddress: MPL_CORE_ORACLE_EXAMPLE_PROGRAM_ID,
+        lifecycleChecks: {
+          burn: [CheckResult.CAN_REJECT],
+        },
+        pda: {
+          type: 'PreconfiguredOwner',
+        },
```
Comment (danenbm) [inline clients/js/test/externalPlugins/oracle.test.ts:1430]: Technically if they chose plain address it wouldn't be a pda.  So we can either ignore that minor use case or rename this to something like `extraAccounts`

File: clients/js/test/externalPlugins/oracle.test.ts [hunk 2]
```diff
@@ -1786,5 +2170,195 @@ test('it can use one fixed address oracle to deny transfer when a second oracle
...
+          create: [CheckResult.CAN_REJECT],
+          update: [CheckResult.CAN_REJECT],
+          transfer: [CheckResult.CAN_REJECT],
+          burn: [CheckResult.CAN_REJECT],
+        },
+        baseAddress: oracleSigner.publicKey,
+      },
+    ],
+  }).sendAndConfirm(umi);
+
```
Comment (danenbm) [inline clients/js/test/externalPlugins/oracle.test.ts:2323]: Nit: I know I wrote this test and didn't put an assert here but now that we have the externalPlugin assert i think would be good to assert the intial oracle config after create.

File: clients/js/src/instructions/approvePluginAuthority.ts [hunk 3]
```diff
@@ -1,39 +1,25 @@
...
-) => {
-  if (isExternalPluginType(plugin)) {
-    // TODO implement this
-    // return approveExternalPluginAuthorityV1(context, {
-    //   ...args,
-    //   key: externalPluginKeyToBase(plugin as ExternalPluginKey),
-    // });
-  }
-
-  return approvePluginAuthorityV1(context, {
```
Comment (danenbm) [inline clients/js/src/instructions/approvePluginAuthority.ts:34]: Not in scope for this PR, but for `updateDelegate` we have the currently unused `additinalDelegates` array, and I'm guessing we will be approving those delegates via an `updatePlugin` not `approvePluginAuthority` right?

**Review by danenbm — APPROVED** [review]
Approved but with couple small comments

File: clients/js/test/externalPlugins/oracle.test.ts (hunk 1 above)
Comment (nhanphan) [reply clients/js/test/externalPlugins/oracle.test.ts:1430]: agree - let's do that in a separate PR

File: clients/js/test/externalPlugins/oracle.test.ts (hunk 2 above)
Comment (nhanphan) [reply clients/js/test/externalPlugins/oracle.test.ts:2323]: fixed

**Pull request merged**